### ISSUE TITLE
Problem inserting certain Decimal values

### ISSUE THREAD

**Issue by tombech** [mkleehammer/pyodbc#491]
I have a table with this definition (Azure SQL Server 12.0.2000.8):

```
CREATE TABLE [dbo].[Test](
[Id] [int] IDENTITY(1,1) NOT NULL,
[Foo [decimal](28, 22) NOT NULL,
CONSTRAINT [PK_Test] PRIMARY KEY CLUSTERED
(
[Id] ASC
)WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON)
)
```

If I try to insert this value into it:

`INSERT INTO [dbo].[Test] ([Foo]) OUTPUT inserted.[Id] VALUES (?)'] [parameters: (Decimal('0.0123456789012345678901'))]
`
I get this error:
`[HY104] [Microsoft][ODBC Driver 17 for SQL Server]Invalid precision value (0) (SQLBindParameter)`

If I use `Decimal('1.0123456789012345678901')` or `Decimal('0.1123456789012345678901')` it works fine. So only Decimal numbers with '0' on both sides of the decimal separator seem to be affected. I'm not familiar with the code but perhaps it computes the scale and/or precision wrong in the original case and end up zero-padding the resulting string buffer with too many '0's, and thus exceed the scale/precision of the target column?

**Comment by v-makouz** [comment]
This appears to be a bug in pyODBC calculation for precision. I'm looking at what would make the most optimal solution.

**Comment by v-makouz** [comment]
PR with the fix: https://github.com/mkleehammer/pyodbc/pull/494

**Comment by mkleehammer** [comment]
Fix from v-makouz merged into 4.0.25.  Thanks!
